src/core/exec-invoke.c

```diff
@@ -2280,6 +2280,56 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int is_idmapping_supported(const char *path) {
+        _cleanup_close_ int mount_fd = -EBADF, userns_fd = -EBADF, dir_fd = -EBADF;
+        _cleanup_free_ char *uid_map = NULL, *gid_map = NULL;
+        int r;
+
+        assert(path);
+
+        dir_fd = open(path, O_RDONLY|O_CLOEXEC|O_DIRECTORY|O_NOFOLLOW);
```

Comment:
why O_DIRECTORY?

Reply:
we need it because the path has to be a mountpoint -> then also has to be a directory

Reply:
not really, any inode can be a mount point, not just directories. I think the function here should be generic and not make such a restriction needlessly.